dvc/objects/db/base.py

```diff
@@ -399,8 +399,15 @@ def hashes_exist(self, hashes, jobs=None, name=None):
         # hashes_exist() (see ssh, local)
         assert self.fs.TRAVERSE_PREFIX_LEN >= 2
 
+        # During the tests, for ensuring that the traverse behavior
+        # is working we turn on this option. It will ensure the
+        # list_hashes_traverse() is called.
+        always_traverse = getattr(self.fs, "_ALWAYS_TRAVERSE", False)
+
```

Comment:
It feels a bit weird to amend the core code like that only for the tests. I guess this is a hint for an upcoming refactor of this part of code :slightly_frowning_face:

Reply:
@pmrowla Do you foresee this part being refactored to accommodate the test better? Maybe you have any suggestions?